lib/posix/options/fs.c

```diff
@@ -412,3 +416,18 @@ int mkdir(const char *path, mode_t mode)
 
 	return 0;
 }
+
+/**
+ * @brief Truncate file to specified length.
+ *
+ */
+int ftruncate(int fd, off_t length)
+{
+	struct posix_fs_desc *ptr = NULL;
+
+	ptr = z_get_fd_obj(fd, NULL, EBADF);
+	if (!ptr)
+		return -1;
+
+	return fs_truncate(&ptr->file, length);
```

Comment:
AFAIK fs_truncate may return other values than 0 or -1, so I think this should be handled properly by setting errno appropriately and returning -1?

Reply:
Good catch @kartben. @karthi012 - can you please address this in a follow-up pr? Also, please ensure that fsync is marked as supported in the docs.

Reply:
@kartben Thanks for comments.  @cfriedt, i will address.